### PR TITLE
Intellect v2.0 

### PR DESCRIPTION
**Pull request by novosaleks**
Intellect new project structure

### CODE REVIEW

File: src/components/I-SearchBlock/I-SearchBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const ISearchBlock: React.FC = () => {
+  return (
+    <main>
+      <section className='pt-18'>
+        <div className='text-5xl'>
+          <h1 className='font-semibold'>Інтелект</h1>
+          <h2 className='font-light'>Викладачі та науковці</h2>
+        </div>
+        <p className='max-w-500 pt-4'>
+          Проект об′єднує вчених, викладачів, інженерів та аспірантів університету, які займаються інтелектуальною творчою діяльністю, проводять фундаментальні таприкладні наукові дослідження,
```
Comment (qqlexa) [inline src/components/I-SearchBlock/I-SearchBlock.tsx:14]:
```suggestion
          Проєкт об′єднує вчених, викладачів, інженерів та аспірантів університету, які займаються інтелектуальною творчою діяльністю, проводять фундаментальні та прикладні наукові дослідження,
```

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import { Link } from 'react-router-dom';
+
+import './Header.css'
+import Burger from '../Burger/Burger';
+import React, { useState } from 'react';
+
+const links = [
+  {to: '/', label: 'Пошук'},
+  {to: 'about', label: 'Про проект'},
```
Comment (qqlexa) [inline src/components/Header/Header.tsx:12]:
```suggestion
  {to: 'about', label: 'Про проєкт'},
```

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          <p className='mt-2'>
+            Адреса: Україна, 03056 м.Київ-56, проспект Перемоги, 37
+          </p>
+        </section>
+        <section>
+          <p>
+            Розробник – Конструкторське бюро інформаційних систем
+          </p>
+          <p className='mt-2'>
+            Створено в рамках проекту "Електронний кампус КПІ"
```
Comment (qqlexa) [inline src/components/Footer/Footer.tsx:25]:
```suggestion
            Створено в рамках проєкту "Електронний кампус КПІ"
```

File: src/components/pages/I-About/I-About.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import SectionTitle from '../../common/SectionTitle';
+
+import aboutImg from '../../../assets/img/sections/about.png';
+import React from "react";
+
+const IAbout: React.FC = () => {
+  return (
+    <InfoBlock sectionImg={aboutImg}>
+      <article>
+        <SectionTitle>Про проект</SectionTitle>
```
Comment (qqlexa) [inline src/components/pages/I-About/I-About.tsx:11]:
```suggestion
        <SectionTitle>Про проєкт</SectionTitle>
```

File: src/components/I-SearchBlock/I-SearchBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+          впроваджують отримані результати в виробництво, займаються навчальною, методичною і організаційною роботою.
```
Comment (qqlexa) [inline src/components/I-SearchBlock/I-SearchBlock.tsx:15]:
```suggestion
          впроваджують отримані результати у виробництво, займаються навчальною, методичною і організаційною роботою.
```

File: src/components/pages/I-About/I-About.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        <p className='mt-4'>
+          Об′єднує вчених, викладачів, інженерів та аспірантів університету, які займаються інтелектуальною творчою діяльністю,
+          проводять фундаментальні та прикладні наукові дослідження, впроваджують отримані результати в виробництво, з
```
Comment (qqlexa) [inline src/components/pages/I-About/I-About.tsx:14]:
```suggestion
          проводять фундаментальні та прикладні наукові дослідження, впроваджують отримані результати у виробництво, з
```

File: src/components/pages/Search/Search.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
+import RoutePointer from '../../RoutePointer/RoutePointer';
+import Alphabet from '../../Alphabet/Alphabet';
+import InputField from '../../InputField/InputField';
+import SearchGrid from '../../common/SearchGrid';
+import ITeacherCard from '../../I-TeacherCard/I-TeacherCard';
+
+import {Location, useLocation} from 'react-router-dom';
```
Comment (qqlexa) [inline src/components/pages/Search/Search.tsx:7]: On the other hand, we have  `import { RouterProvider } from 'react-router-dom';` let's provide the same behavior with spaces between brackets `{ }`

I would choose spaces `import { Location, useLocation } from 'react-router-dom';`. What do you think?

File: src/components/pages/Contacts/Contacts.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+const Contacts: React.FC = () => {
+  return (
+    <InfoBlock sectionImg={sectionImg}>
+      <div>
+      <SectionTitle>Контакти</SectionTitle>
+        <div className='mt-4'>
+          <div className='leading-6'>
+            <div className="font-semibold mb-1">Інформаційна підтримка</div>
+            <div>тел.: <a href='tel:380444549845'>+38 (044) 454 98 45</a></div>
+            <div>факс: <a href='tel:380444549845'>+38 (044) 454 98 45</a></div>
```
Comment (qqlexa) [inline src/components/pages/Contacts/Contacts.tsx:16]:
```suggestion
            <div>тел.: <a href='tel:380444549845'>+380 (44) 454 98 45</a></div>
            <div>факс: <a href='tel:380444549845'>+380 (44) 454 98 45</a></div>
```

File: src/components/Footer/Footer.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+const Footer: React.FC<Props> = ({logoSrc}) => {
+  return (
+    <footer className='bg-primary text-white text-sm'>
+      <div className='wrapper flex sm:flex-row sm:text-left text-center flex-col justify-between py-9 gap-9'>
+        <img className='max-w-120 mx-auto' src={logoSrc} alt='logo' />
+        <section>
+          <p>
+            Національний технічний університет України"Київський політехнічний інститут імені Ігоря Сікорського" © 1998-2023
```
Comment (qqlexa) [inline src/components/Footer/Footer.tsx:14]: Can we set dynamic current_year instead `2023`?

File: src/components/Footer/Footer.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+          <p>
+            Розробник – Конструкторське бюро інформаційних систем
+          </p>
+          <p className='mt-2'>
+            Створено в рамках проекту "Електронний кампус КПІ"
+          </p>
+        </section>
+        <section>
+          <p>
+            Використання матеріалу сайту тільки з обов’язковим посиланням на intellect.kpi.ua
```
Comment (qqlexa) [inline src/components/Footer/Footer.tsx:30]:
```suggestion
            Використання матеріалу сайту тільки з обов’язковим посиланням на <a href="https://intellect.kpi.ua">intellect.kpi.ua</a>
```

File: src/components/I-TeacherSearch/I-TeacherSearch.tsx [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  type: Intellect.SearchMode,
+  placeholder?: string
+}
+
+const ITeacherSearch: React.FC = () => {
+  const [activeTab, setActiveTab] = useState('overall' as Intellect.SearchMode);
+  const navigate = useNavigate();
+
+  const tabs = [
+    {label: 'Загальний пошук спiвробiтникiв', type: 'overall', placeholder: 'Введіть ПІБ особи.. (наприклад: Петро Петров Петрович)'},
```
Comment (qqlexa) [inline src/components/I-TeacherSearch/I-TeacherSearch.tsx:21]: Let's provide the same order in an example like ПІБ:
```suggestion
    {label: 'Загальний пошук спiвробiтникiв', type: 'overall', placeholder: 'Введіть ПІБ особи.. (наприклад: Петров Петро Петрович)'},
```

File: src/components/pages/I-TeacherInfo/I-TeacherInfo.tsx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+      <RoutePointer routePath={route} />
+      <div className='flex flex-col sm:flex-row items-center sm:items-start gap-6 mt-6'>
+        <div>
+          <Avatar img={teacher?.photo} />
+          <ContactBlock contactRecords={teacher?.contactRecords} className='hidden sm:block' />
+        </div>
+        <div className='flex-1 w-full'>
+          <SectionTitle className='text-3xl sm:text-5xl text-center sm:text-left' isPrimary={false}>{teacher?.fullName}</SectionTitle>
+          <div className='flex gap-3 mt-5 justify-center sm:justify-start overflow-x-auto'>
+            {/*TODO*/}
```
Comment (qqlexa) [inline src/components/pages/I-TeacherInfo/I-TeacherInfo.tsx:112]: Provide detailed comment with TODO

File: src/components/pages/Contacts/Contacts.tsx (hunk 8 above)
Comment (qqlexa) [reply src/components/pages/Contacts/Contacts.tsx:16]: Detailed information - 
https://uk.wikipedia.org/wiki/%D0%A2%D0%B5%D0%BB%D0%B5%D1%84%D0%BE%D0%BD%D0%BD%D1%96_%D0%BA%D0%BE%D0%B4%D0%B8_%D0%A3%D0%BA%D1%80%D0%B0%D1%97%D0%BD%D0%B8

File: src/api/teacher.ts [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+        })
+    } catch (e: any) {
+        throw new Error(e)
+    }
+
+    return resultObj;
+}
+
+export const getTeacherByTeacherId = (teacherId: string): Promise<Intellect.Teacher> => {
+    return Http.get('https://dev-api.campus.cloud.kpi.ua/account/public/' + teacherId);
```
Comment (qqlexa) [inline src/api/teacher.ts:66]: Can we set this domain as environment variable?

File: src/api/index.ts [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios';
+
+const Http = axios.create({
+    baseURL: 'https://dev-api.campus.cloud.kpi.ua/intellect/',
```
Comment (qqlexa) [inline src/api/index.ts:4]: Can we set this domain as environment variable?

File: .github/workflows/workflow.yml [hunk 15]
```diff
@@ -3,7 +3,8 @@ name: Build intellect.kpi.ua
 on:
   push:
     branches:
-      - master
+#      - master
```
Comment (qqlexa) [inline .github/workflows/workflow.yml:6]: Remove commented code before merging to main

**Comment by ernado-x** [comment]
@qqlexa  I suggest you create your pull request into the branch _intellect-new_ if you please.

**Comment by qqlexa** [comment]
> @qqlexa  I suggest you create your pull request into the branch _intellect-new_ if you please.

Sure. I would fix comments and we merge changes to _intellect-new_?

**Comment by ernado-x** [comment]
> > @qqlexa  I suggest you create your pull request into the branch _intellect-new_ if you please.
> 
> Sure. I would fix comments and we merge changes to _intellect-new_?

Yes

**Pull request merged**